### PR TITLE
update code related to 'columns' tag for index definition

### PR DESCRIPTION
**Pull request by skywhat**
1. update dependency packages such as latest dosa-idl
2. cover corner cases
   1)  fields in columns tag must refer to a column
   2) a column cannot be used twice
3. update dosa schema dump feature due to flexible secondary index
4. update Thrift related function in /yarpc

### CODE REVIEW

File: schema/cql/cql.go [hunk 1]
```diff
@@ -55,15 +55,20 @@ func typeMap(t dosa.Type) string {
...
 var cqlCreateTableTemplate = template.Must(template.
 	New("cqlCreateTable").
 	Funcs(map[string]interface{}{"typeMap": typeMap}).
 	Funcs(map[string]interface{}{"uniqueKey": uniqueKey}).
+	Funcs(map[string]interface{}{"isSelectAllCreatingView": isSelectAllCreatingView}).
 	Parse(`create table "{{.Name}}" ({{range .Columns}}"{{- .Name -}}" {{ typeMap .Type -}}, {{end}}primary key {{ .Key }});
 {{- range $name, $indexdef := .Indexes }}
 create materialized view "{{- $name -}}" as
-  select * from "{{- $.Name -}}"
+  select {{if isSelectAllCreatingView $indexdef.Columns}}*{{else}}{{range $colnum, $col := $indexdef.Columns}}{{if $colnum}}, {{end}}"{{$col}}"{{end}}{{end}} from "{{- $.Name -}}"
```
Comment (phliar) [inline schema/cql/cql.go]: What is the test that tells us there is no error in this line?

File: schema/cql/cql_test.go [hunk 2]
```diff
@@ -71,7 +80,18 @@ create materialized view "i2" as
   where "int64type" is not null
   primary key (int64type, booltype ASC);`,
 		},
-		// TODO: Add more test cases
```
Comment (phliar) [inline schema/cql/cql_test.go]: We no longer need to add more test cases?

**Comment by phliar** [comment]
Are these changes related, or separate? If they are separate changes we should not combine them in one diff.

**Comment by skywhat** [comment]
> Are these changes related, or separate? If they are separate changes we should not combine them in one diff.
I think they could be treated as related cause these changes are needed due to the introduction of our new flexible secondary index feature.

File: schema/cql/cql_test.go (hunk 2 above)
Comment (skywhat) [reply schema/cql/cql_test.go]: kk. put it back

File: schema/cql/cql.go (hunk 1 above)
Comment (skywhat) [reply schema/cql/cql.go]: https://github.com/uber-go/dosa/blob/index/schema/cql/cql_test.go#L84

**Pull request closed without merging**